sdk/storage/azure-storage-blob/azure/storage/blob/blob_client.py

```diff
@@ -289,12 +291,20 @@ def generate_shared_access_signature(
         :param str content_type:
             Response header value for Content-Type when resource is accessed
             using this shared access signature.
+        :param ~azure.storage.blob._shared.models.UserDelegationKey user_delegation_key:
+            Instead of an account key, the user could pass in a user delegation key.
+            A user delegation key can be obtained from the service by authenticating with an AAD identity;
+            this can be accomplished by calling get_user_delegation_key.
+            When present, the SAS is signed with the user delegation key instead.
         :return: A Shared Access Signature (sas) token.
         :rtype: str
         """
-        if not hasattr(self.credential, 'account_key') or not self.credential.account_key:
-            raise ValueError("No account SAS key available.")
-        sas = BlobSharedAccessSignature(self.credential.account_name, self.credential.account_key)
+        if user_delegation_key is not None:
+            sas = BlobSharedAccessSignature(self.account_name, user_delegation_key=user_delegation_key)
```

Comment:
>self.account_name [](start = 44, length = 17)

use self.credential.account_name instead?

Reply:
credential could be different type, I think this is only working for SharedKeyCredential

Reply:
Correct - we can only guarantee that we have the `account_name` if the user is authenticating with SharedKeyCredentials

Reply:
based on our discussion offline, we want to add a parameter account_name to generate_shared_access_signature method, so when they are using oauth token as credential, they have to provide account_name to generate sas token.